### ISSUE TITLE
Rules voting

### ISSUE THREAD

**Issue by brokenmass** [MailOnline/eslint-config-mailonline#8]
Placeholder for specific rules votations

**Comment by brokenmass** [comment]
**arrow-body-style:**
vote

👍 : if you want the as-needed option ( developer HAS to follow the rule of not using bracket around body if there's only a single return instruction )

```
[1, 2, 3, 4].map((element) => element * element);
```

👎 : if you want the rule disabled (so developer will choose the style based on the specific case )

```
[1, 2, 3, 4].map((element) => element * element);
OR
[1, 2, 3, 4].map((element) => { 
  return element * element;
});
```

❤️ : if you want "always", (developer will have to ALWAYS use brackets around arrow function body) 

```
[1, 2, 3, 4].map((element) => { 
  return element * element;
});
```

**Comment by brokenmass** [comment]
**allowSingleLine:**
vote:
-1 : if you want to allows the opening and closing braces for a block to be on the same line
```
if(true) { return 'lol'; }
```

+1 : if you want opening and closing braces for a block to always be on different lines
```
if(true) { 
  return 'lol'; 
}
```

**Comment by gajus** [comment]
❤️ : if you want "always"

:-)

**Comment by brokenmass** [comment]
👍 VOTES

**Comment by gajus** [comment]
People voting ❤️ get a free beer on me!

(Please note: Vote fairly. A **FREE** beer, must not effect your vote.)

**Comment by ikhemissi** [comment]
**func-style:** 
vote

👍  : if you want function "[declarations](http://eslint.org/docs/rules/func-style#declaration)"

``` javascript
function foo () {
  // do something
}
```

👎  : if you want "[expressions](http://eslint.org/docs/rules/func-style#expression)"

``` javascript
const foo = () => {
  // do something
};
```

😕  (confused): if you don't want to enforce any of the above

**Comment by gajus** [comment]
@ikhemissi Using one vote per person per poll would make the results more clear. Otherwise, there can be no majority.

**Comment by claudiorodriguez** [comment]
Max empty lines #9 :
- :+1: for keeping it as it is (max 2 lines)
- :-1: for changing it to max 1 line

**Comment by gajus** [comment]
> Max empty lines #9 :

Just for the record, I am updating Canonical to use 1 max line.

I have looked at couple code bases to analyse how the empty lines are used. I did not find any valid cases for using two spaces instead of one. In fact, most places where I have found two empty lines, appear to be there by mistake.

**Comment by gajus** [comment]
> arrow-body-style:
> vote

I am not loving the "as-needed" option.

Example:

Before:

``` js
const filterTopicsByGuide = (topics, guide) => {
  return topics
    .filter((topic) => {
      const firstCharacter = topic.name.slice(0, 1).toLowerCase();

      if (guide.length === 1) {
        return firstCharacter === guide;
      }

      // A special catch all case ('0-9').
      return !/^[a-z]$/.test(firstCharacter);
    });
};

// eslint-disable-next-line import/no-mutable-exports
let GuideListComponent;

GuideListComponent = ({topics}) => {
  return <ol styleName='guide-list'>
    {indexGuide.map((guide) => (
      <li>
        <a name={'topic-index-guide-' + guide} />
        <div styleName='heading'>
          <span styleName='guide'>{guide}</span>
        </div>
        <TopicListComponent topics={filterTopicsByGuide(topics, guide).sort()} />
      </li>
    ))}
  </ol>;
};
```

After:

``` js
const filterTopicsByGuide = (topics, guide) =>
  topics
    .filter((topic) => {
      const firstCharacter = topic.name.slice(0, 1).toLowerCase();

      if (guide.length === 1) {
        return firstCharacter === guide;
      }

      // A special catch all case ('0-9').
      return !/^[a-z]$/.test(firstCharacter);
    });

// eslint-disable-next-line import/no-mutable-exports
let GuideListComponent;

GuideListComponent = ({topics}) =>
  <ol styleName='guide-list'>
    {indexGuide.map((guide) => (
      <li>
        <a name={'topic-index-guide-' + guide} />
        <div styleName='heading'>
          <span styleName='guide'>{guide}</span>
        </div>
        <TopicListComponent topics={filterTopicsByGuide(topics, guide).sort()} />
      </li>
    ))}
  </ol>
```

It makes the code look cleaner (I guess), but it is frustrating to keep needing change the code between having not having the brackets every time you need to add `debugger;` statement or `console.log` or whatever else.

Furthermore, I'd argue that this "cleaner" look becomes harder to read as the nesting increases.

Just expressing my concern.

For these reasons, I'd prefer "always" option.

**Comment by gajus** [comment]
Another example where no braces make the code harder to read:

``` js
{
  loader: 'postcss-loader',
  options: {
    plugins: () => [
      // eslint-disable-next-line global-require
      require('autoprefixer')({
        browsers: [
          'last 2 versions'
        ]
      })
    ]
  }
},
```

and another:

``` js
return topics.map((topic) => ({
  ...topic,
  id: String(id++),
  url: 'http://google.com'
}));
```

I think that braceless expressions make sense when invoking a function, e.g.

``` js
foo => make(foo)
```

Thats cool.

However, as soon as you have a multi line spanning code block or array/ object, it becomes inconsistent and hard to follow.

**Comment by claudiorodriguez** [comment]
I agree with Gajus. I would say that on like 5% of the cases where you would use an arrow function, it does make the code more readable (e.g. `.map(number => number * 2)`), but on the vast majority of commonplace code, it makes things harder to follow, and modifying things sometimes needs an extra annoying step.

**Comment by claudiorodriguez** [comment]
Closing this as team consensus has been reached.
